### PR TITLE
Implemented Modbus Function code FC23 client function.

### PR DESCRIPTION
**Pull request by Okoyl**
I have implemented FC23 support, it allows writing and reading multiple registers on a single transmission.

### CODE REVIEW

File: src/tcp.rs [hunk 1]
```diff
@@ -216,6 +216,43 @@ impl Transport {
         self.write(&mut buff)
     }
 
+    fn write_read_multiple(&mut self, fun: &Function) -> Result<Vec<u8>> {
+        let (write_addr, write_quantity, write_values, read_addr, read_quantity) = match *fun {
```
Comment (hirschenberger) [inline src/tcp.rs:220]: You can simplify this with a single: 

```rust 
if let Function::WriteReadMultipleRegisters(...) = *fun {
...
} else {
...
}```

**Review by hirschenberger — COMMENTED** [review]
Hello, thank you for that PR. Can you also please add `WriteReadMultiple` tests to `tests/lib.rs` and check them with:

```
cargo test --features modbus-server-tests
```

File: src/tcp.rs (hunk 1 above)
Comment (Okoyl) [reply src/tcp.rs]: > You can simplify this with a single:
> 
> ```rust
> if let Function::WriteReadMultipleRegisters(...) = *fun {
> ...
> } else {
> ...
> }``` 
> ```

Good Idea, I just commited it.

**Comment by Okoyl** [comment]
> Hello, thank you for that PR. Can you also please add `WriteReadMultiple` tests to `tests/lib.rs` and check them with:
> 
> ```
> cargo test --features modbus-server-tests
> ```

I added a simple test that checks it twice, it passes the test.

File: src/tcp.rs [hunk 2]
```diff
@@ -216,6 +216,43 @@ impl Transport {
...
+                            Transport::validate_response_code(&buff, &reply)?;
+                            Transport::get_reply_data(&reply, expected_bytes)
+                        }
+                        Err(e) => Err(Error::Io(e)),
+                    }
+                }
+                Err(e) => Err(Error::Io(e)),
+            }
+        } else {
+            return Err(Error::InvalidFunction);
```
Comment (hirschenberger) [inline src/tcp.rs:252]: That `return` statement is not neeed

**Pull request merged**